Makefile

```diff
@@ -604,6 +605,26 @@ $(SBOM): $(ROOTFS_TAR) | $(INSTALLER)
 	rm -rf $(TMP_ROOTDIR)
 	$(QUIET): $@: Succeeded
 
+$(GENERATE_METADATA): $(ROOTFS_TAR) | $(INSTALLER)
+	$(QUIET): $@: Begin
+	$(eval TMP_ROOTDIR := $(shell mktemp -d))
+	# this is a bit of a hack, but we need to extract the rootfs tar to a directory, and it fails if
+	# we try to extract character devices, block devices or pipes, so we just exclude the dir.
+	# when syft supports reading straight from a tar archive with duplicate entries,
+	# this all can go away, and we can read the rootfs.tar
+	# see https://github.com/anchore/syft/issues/1400
+	tar xf $< -C $(TMP_ROOTDIR) --exclude "dev/*"
+	bash tools/get-alpine-pkg-source.sh -v -s /tmp/sources/alpine -e $(TMP_ROOTDIR) 
+	bash tools/get-kernel-source.sh -v -s /tmp/sources/kernel 
+	echo $(shell $(ROOTDIR))
+	for i in $$(find . -name 'go.sum'); \
+	do								 		   \
+		go-sources-and-licenses sources -d $$(dirname $$i) --recursive --out /tmp/sources/go; \
+	done;	
+	tar cvf - /tmp/sources/ | gzip -9 - > ./metadata.tar.gz
```

Comment:
We place the other output of the build (like the rootfs etc) in the dist/amd64/... directory. I think it makes sense to put this output there (and name it something like collected_sources).

Reply:
Yes. Make sure it is not in `installed/`, so it is not included in the final eve image and does not bloat it, but `dist/<arch>/` is the right place.

Also, you can simplify this line to:

```
tar zcvf $(DIST)/metadata.tar.gz -C /tmp/sources/ .
```

(that `$(DIST)` probably is incorrect, but the right idea)

Reply:
Also, can we clean up `/tmp/sources` and use something for each build, that gets cleaned up later? See how we did it for sbom:

```makefile
SBOM=$(ROOTFS).spdx.json
sbom: $(SBOM)
$(SBOM): $(ROOTFS_TAR) | $(INSTALLER)
	$(QUIET): $@: Begin
	$(eval TMP_ROOTDIR := $(shell mktemp -d))
	# this is a bit of a hack, but we need to extract the rootfs tar to a directory, and it fails if
	# we try to extract character devices, block devices or pipes, so we just exclude the dir.
	# when syft supports reading straight from a tar archive with duplicate entries,
	# this all can go away, and we can read the rootfs.tar
	# see https://github.com/anchore/syft/issues/1400
	tar xf $< -C $(TMP_ROOTDIR) --exclude "dev/*"
	docker run -v $(TMP_ROOTDIR):/rootdir:ro $(SYFT_IMAGE) -o spdx-json /rootdir > $@
	rm -rf $(TMP_ROOTDIR)
	$(QUIET): $@: Succeeded
```

So we probably could do:

```makefile
COLLECTED_SOURCES=$(DIST)/sources.tar.gz
collected_sources: $(COLLECTED_SOURCES)
$(COLLECTED_SOURCES): $(ROOTFS_TAR) | $(INSTALLER)
	$(QUIET): $@: Begin
	$(eval TMP_SOURCES := $(shell mktemp -d))
       # do whatever you need for sources here
       # when done:
	tar -zcvf $@ -C $(TMP_SOURCES) .
	rm -rf $(TMP_SOURCES)
	$(QUIET): $@: Succeeded
```

Reply:
cleanup implemented. Copied the artifact to the dist folder